scipy/special/basic.py

```diff
@@ -754,20 +754,101 @@ def riccati_yn(n, x):
 
 
 def erfinv(y):
-    """Inverse function for erf.
+    """Inverse of the error function erf.
+
+    Computes the inverse of the error function.
+
+    Inverse of the error function satisfies erf(erfinv(x)) = x.
+
+    Parameters
+    ----------
+    y : ndarray
+        Argument at which to evaluate. Domain: (-1, 1)
+
+    Returns
+    -------
+    erfinv : ndarray
+        The inverse of erf of y, element-wise
+
+    Examples
+    --------
+    1) evaluating a float number
+
+    >>> from scipy import special
+    >>> special.erfinv(0.5)
+    0.4769362762044698
+
+    2) evaluating a ndarray
+
+    >>> from scipy import special
+    >>> y = np.linspace(-1.0, 1.0, num=10)
+    >>> special.erfinv(y)
+    array([       -inf, -0.86312307, -0.5407314 , -0.30457019, -0.0987901 ,
+            0.0987901 ,  0.30457019,  0.5407314 ,  0.86312307,         inf])
+
     """
     return ndtri((y+1)/2.0)/sqrt(2)
 
 
 def erfcinv(y):
-    """Inverse function for erfc.
+    """Inverse of the complementary error function erfc.
+
+    Computes the inverse of the complementary error function erfc.
+
+    It is related to inverse of the error function by erfcinv(1-x) = erfinv(x)
+
+    Parameters
+    ----------
+    y : ndarray
+        Argument at which to evaluate. Domain: (0, 2)
+
+    Returns
+    -------
+    erfcinv : ndarray
+        The inverse of erfc of y, element-wise
+
+    Examples
+    --------
+    1) evaluating a float number
+
+    >>> from scipy import special
+    >>> special.erfcinv(0.5)
+    0.4769362762044698
+
+    2) evaluating a ndarray
+
+    >>> from scipy import special
+    >>> y = np.linspace(0.0, 2.0, num=11)
+    >>> special.erfcinv(y)
+    array([        inf,  0.9061938 ,  0.59511608,  0.37080716,  0.17914345,
+            -0.        , -0.17914345, -0.37080716, -0.59511608, -0.9061938 ,
+                  -inf])
+
     """
     return -ndtri(0.5*y)/sqrt(2)
 
 
 def erf_zeros(nt):
     """Compute nt complex zeros of error function erf(z).
 
```

Comment:
Which zeros is `erf_zeros` returning?  The first `nt` zeros? Ordered how? Does it return all zeros, or just some special subset of the zeros?  I'm wondering if it just returns zeros in the first quadrant,  and that if `z` is a zero, then so is `-z` and perhaps also `+/- conj(z)`? If so, stating that would be useful to understand what it is doing.

Reply:
Is it the case that `erf(-z) = erf(z)`  and `erf(conj(z)) = conj(erf(z))`?  If so, then each zero generates  a total of 4 zeros  `{z, -z, iz, -iz}`.
Perhaps the following?

"""Compute the first `nt` zero in the first quadrant, ordered by absolute value.  

Zeros in the other quadrants can be obtained by using the symmetries `erf(-z) = erf(z)`  and `erf(conj(z)) = conj(erf(z))`.
"""

Reply:
In that case, to calculate the location of the root of `erf` in complex region, it is not using the `erf` itself.
As I previously mentioned before [comment](https://github.com/scipy/scipy/pull/9055#discussion_r204261054), the function `w(z)` defined by `w(z) = exp(-z^2)*erfc(-z)` is used for calculating.

In general, `erf` in complex plane has a y symmetry.
I also attach a reference site for calculating the root loc of `erf` below.

Reply:
My interpretation of the referenced paper ["Complex Zeros of the Error Function and of the Complementary Error Function" by Fettis, Caslin & Kramer] is that it was mainly concerned with deriving zeros of `erfc(z)` using `w(z)`, not zeros of `erf(z)`.  
DLMF https://dlmf.nist.gov/7.13#i lists the zeros of `erfc(z)` with symmetry `{z, -z, conj(z), -conj(z)}`.   The routine `CERZO` in `special/specfun/specfun.f` appears to use the first two terms of that equation 7.13.1 to generate an estimate of the n-th root of `erfc(z)` and then uses Newton's method to refine it.   I don't think that `w(z)` appears anywhere in this calculation.

Reply:
Let's put together the current conversation.
- the definition of `w` in `erfinv()`, `erfcinv()` : complex number
- there is no definition in `erf_zeros()`

After reading the contents of the conversation again, it seems like the subject of the story was different.

I have not written any description in `erf_zeros()`. 
As you already know, the `w(z)` of the comment is a description of the referenced paper.
Actually, the `erf_zeros()` is using the iteration method for calculating a zeros by `CERZO` routine.

There are four complex numbers for each zeros point of `erf()` 
because the direction of the solution is xy-symmetry as you say.

The y-symmetry I have mentioned is that the `erf()` function itself is y-symmetric in the complex domain,
and the solution passing zeros is xy-symmetry.

Back to the first question
> Which zeros is `erf_zeros` returning? The first `nt` zeros? Ordered how? Does it return all zeros, or just some special subset of the zeros? I'm wondering if it just returns zeros in the first quadrant, and that if `z` is a zero, then so is `-z` and perhaps also `+/- conj(z)`? If so, stating that would be useful to understand what it is doing.

As you already know the answer, there are four positions {z, -z, conj(z), -conj(z)} because xy-symmetry

Reply:
@akahard2dj Your PR adds a lot to really useful documentation to `erf_zeros` with `Parameters`, `Returns` and `Examples` sections.  The only thing missing from the docstring now is a usable explanation of what the function does.  I had to dig quite a while to find the answers to my questions.   My suggestion is still to replace the first sentence in the docstring with
"""Compute the first `nt` zero in the first quadrant, ordered by absolute value.

Zeros in the other quadrants can be obtained by using the symmetries `erf(-z) = erf`(z) and `erf(conj(z)) = conj(erf(z))`.
"""

Reply:
Sure. No problem.

I have also worried about the content for adding a description of `erf_zeros()`.
Your suggestion very good and valuable explanation for `erf_zeros()`.

I will add a docstring yours in my commit.
Thank you so much @pvanmulbregt.